libraries/c_sdk/standard/mqtt/src/iot_mqtt_serializer_deserializer_wrapper.c

```diff
@@ -395,9 +422,6 @@ IotMqttError_t _IotMqtt_subscribeSerializeWrapper( const IotMqttSubscription_t *
         *pPacketIdentifier = packetId;
     }
 
-    /* Free allocated memory as the packet was serialized. */
-    IotMqtt_FreeMessage( subscriptionList );
```

Comment:
It looks like this introduces another memory leak by not freeing the subscription list on success? I believe restoring this line and deleting lines 385 and 412 would alway free the subscription list